kani-compiler/src/kani_compiler.rs

```diff
@@ -166,7 +167,9 @@ impl KaniCompiler {
                 CompilationStage::CodegenNoStubs { .. } => {
                     unreachable!("This stage should always run in the same session as Init");
                 }
-                CompilationStage::CodegenWithStubs { target_harness, all_harnesses, .. } => {
+                CompilationStage::CodegenWithStubs { target_harnesses, all_harnesses, .. } => {
+                    assert!(!target_harnesses.is_empty());
```

Comment:
Is that an invariant? Should we provide an error message?

Reply:
Yes, it is an invariant. I can add a message... we don't always bother though. Crashes in the compiler usually come with a user friendly message saying that it is a bug and a link to report the issue.

The message is mostly for our consumption, usually used to narrow down the problem. In cases like this, I feel like we're just stating the obvious, something like: "Expected at least one harness to be processed, but found none".